site/src/components/WasThisHelpful.js

```diff
@@ -0,0 +1,162 @@
+                <MultiLineInput label={'Feedback'} changeFunction={handleFeedback} placeholder={feedbackPlaceholder} value={userFeedback}/>
+                <SingleLineInput label={'Email'} changeFunction={handleEmail} placeholder={'Optional'} value={userEmail} note={emailNote} valid={isEmailValid}/>
+                <Alert type={'info'} text={'Your email will only be used to contact you regarding this feedback.'}/>
+                <div className="feedback-buttons">
+                    <InteractiveButton type={'secondary'} onClick={submitFeedback} isDisabled={awaitingResponse} isLoading={awaitingResponse} text={'Submit'}/>
+                    <InteractiveButton type={'destructive'} onClick={cancelFeedback} isDisabled={awaitingResponse} text={'Cancel'}/>
+                </div>
+                {requestError && <Alert type={'error'} text={requestErrorMessage}/>}
+            </div>
+        )
+    };
+
+    const feedbackSubmitted = () => {
+        const thanks = 'Thanks! Your feedback has been submitted.';
+        return(
+            <div className="feedback-submitted">{userEmail ? `${thanks} We will contact you at ${userEmail}` : thanks}</div>
+        )
+    };
+
+    const handleFeedback = (e) => {
+        setUserFeedback(e.target.value);
+    };
+
+    const handleEmail = (e) => {
+        setUserEmail(e.target.value);
+        setIsEmailValid(true);
+        setEmailNote('');
+    }
+
+    const submitHelpful = (isHelpful) => {
+        setFeedbackQuality(isHelpful);
+        setQuestionOpacity(0);
+        setFeedbackPlaceholder(isHelpful=='good' ? 'How was this page helpful? (Optional)' : 'What can we improve? (Optional)');
+        
+        setTimeout(() => {
+            setIsHelpfulSubmitted(true);
+            setFeedbackOpacity(1);
+        }, transitionTime)
+    };
+
+    const cancelFeedback = () => {
+        setFeedbackOpacity(0);
+
+        setTimeout(() => {
+            setIsHelpfulSubmitted(false);
+            setQuestionOpacity(1);
+            setIsEmailValid(true);
+            setEmailNote('');
+            setUserFeedback('');
+            setUserEmail('');
+        }, transitionTime)
+    };
+
+    const submitFeedback = async () => {
+        setRequestError(false);
+
+        const emailRegex = /^\S+@\S+$/;
+        if(userEmail && !emailRegex.test(userEmail)) {
+            setIsEmailValid(false);
+            setEmailNote('Please enter a valid email address.');
+            return;
+        }
+
+        const feedbackBody = {
+            timestamp: new Date(Date.now()),
+            pageId: pageId,
+            feedbackType: feedbackQuality,
+            feedbackString: userFeedback,
+            userEmail: userEmail
+        };
+        setAwaitingResponse(true);
+
+        try {
+            setTimeout(() => controller.abort(), 20000);
+            const response = await fetch('https://eowxoldwz4d7syt.m.pipedream.net', {method: 'POST', body: JSON.stringify(feedbackBody), signal: controller.signal});
```

Comment:
Can we set this URL as a variable in the docusaurus config file?

Reply:
yes, but is there a reason to? This is the only place it'll be used